bundles/core/org.eclipse.smarthome.core.thing.test/src/test/groovy/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServiceOSGiTest.groovy

```diff
@@ -388,6 +388,41 @@ final class FirmwareUpdateServiceOSGiTest extends OSGiTest {
         assertFirmwareStatusInfoEvent(THING1_UID, upToDateInfo)
     }
 
+    @Test(expected=IllegalArgumentException)
+    void 'assert that cancel throws IllegalArgumentException if there is no firmware update handler for thing'(){
+        firmwareUpdateService.cancelFirmwareUpdate(new ThingUID("dummy:thing:withoutHandler"))
+    }
+
+    @Test(expected=NullPointerException)
+    void 'assert that cancel throws NullPointerExceptions if the given thing id is null'(){
+        firmwareUpdateService.cancelFirmwareUpdate(null)
+    }
+
+    @Test
+    void 'assert that cancel cancels the firmware update using the correct FirmwareUpdateHandler'(){
+        def cancelCalled = false
+        def uid = new ThingUID("dummy", "type", "thing123")
+        def firmwareUpdateHandler = [
+            getThing: {
+                ThingBuilder.create(uid).build()
+            },
+            isUpdatable: { false },
+            cancel: { cancelCalled = true}
+        ] as FirmwareUpdateHandler
+        registerService(firmwareUpdateHandler)
+
+        def service = getService(FirmwareUpdateHandler) {
+            bundleContext.getService(it).equals(firmwareUpdateHandler)
+        }
+        assertThat service, is(not(null))
+
+        firmwareUpdateService.cancelFirmwareUpdate(uid)
```

Comment:
could you pls also assert that no further registered firmware update handler has been cancelled?